### DISCUSSION TITLE
Creating Workflow through dynamically

### DISCUSSION THREAD

**Discussion by makattar** [elsa-workflows/elsa-core#1205] [Q&A] [answered]
When a user submits 
Entity : Employee , Activity : Add => Create a workflow EmployeeAdd Which is having Http Endpoint as /employeeadd and make http response as EmployeeAdd

 Entity : Employee , Activity : Delete=> Create a workflow EmployeeDelete Which is having Http Endpoint as /employeedelete and make http response as EmployeeDelete

I came to this...
           WorkflowDefinition mywf = new WorkflowDefinition();
            mywf.Name = "MYWF1";
            mywf.DisplayName = "MYWF!";
            mywf.Description = "MYWF Description";
            var savedres = await workflowPublisher.SaveDraftAsync(mywf);
            await workflowPublisher.PublishAsync(savedres);

How should i add activities here like Writehttpresponse?

**Comment by sfmskywalker** [comment, marked as answer]
You can add activities to the `mywf` object like this:

```csharp
var writeHttpResponse = new ActivityDefinition
{
   ActivityId = Guid.NewGuid().ToString("N"),
   Type = nameof(WriteHttpResponse),
   Properties = new List<ActivityDefinitionProperty>
   {
      ActivityDefinitionProperty.Literal(nameof(WriteHttpResponse.ContentType), "application/json"),
      ... etc.
   }
}
mywf.Activities.Add(writeHttpResponse);
```

You can find another example here: https://github.com/elsa-workflows/elsa-core/blob/master/src/samples/console/Elsa.Samples.DeclarativeCompositeActivitiesConsole/Program.cs#L27

**Reply by makattar** [reply]
Thank you.
